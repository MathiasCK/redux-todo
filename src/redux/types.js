export const ADD_TODO = 'ADD_TODO';

export const TOGGLE_TODO = 'TOGLE_TODO';

export const EDIT_TODO = 'EDIT_TODO';

export const DELETE_TODO = 'DELETE_TODO';

export const SHOW_COMPLETED = 'SHOW_COMPLETED';

export const SHOW_UNCOMPLETED = 'SHOW_UNCOMPLETED';
