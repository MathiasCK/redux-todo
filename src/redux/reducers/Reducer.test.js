import { ADD_TODO, DELETE_TODO, TOGGLE_TODO } from '../types';
import todoReducer from './todo-reducer';

describe('todoReducer', () => {
  it('should return the initial state', () => {
    expect(todoReducer(undefined, {})).toMatchSnapshot();
  });
  test('adds one item succesfully', () => {
    const action = {
      type: ADD_TODO,
      payload: {
        id: 1234,
        text: 'Write reducer test',
        isComplete: false,
      },
    };
    const initialState = undefined;

    const nextState = todoReducer(initialState, action);

    expect(nextState).toEqual({
      todos: [{ id: 1234, text: 'Write reducer test', isComplete: false }],
    });
  });
  test('adds another item succesfully', () => {
    const action = {
      type: ADD_TODO,
      payload: {
        id: 12345,
        text: 'Write another reducer test',
        isComplete: false,
      },
    };
    const initialState = {
      todos: [{ id: 1234, text: 'Write reducer test', isComplete: false }],
    };

    const nextState = todoReducer(initialState, action);

    expect(nextState).toEqual({
      todos: [
        { id: 12345, text: 'Write another reducer test', isComplete: false },
        { id: 1234, text: 'Write reducer test', isComplete: false },
      ],
    });
  });
  test('marks todo as complete', () => {
    const action = {
      type: TOGGLE_TODO,
      payload: {
        id: 1234,
        text: 'Write another reducer test',
        isComplete: false,
      },
    };
    const initialState = {
      todos: [{ id: 1234, text: 'Write reducer test', isComplete: false }],
    };
    const nextState = todoReducer(initialState, action);

    expect(nextState).toEqual({
      todos: [{ id: 1234, text: 'Write reducer test', isComplete: true }],
    });
  });
  test('deletes todo', () => {
    const action = {
      type: DELETE_TODO,
      payload: 1234,
    };
    const initialState = {
      todos: [
        { id: 12345, text: 'Write another reducer test', isComplete: true },
        { id: 1234, text: 'Write reducer test', isComplete: false },
      ],
    };
    const nextState = todoReducer(initialState, action);

    expect(nextState).toEqual({
      todos: [
        { id: 12345, text: 'Write another reducer test', isComplete: true },
      ],
    });
  });
});
