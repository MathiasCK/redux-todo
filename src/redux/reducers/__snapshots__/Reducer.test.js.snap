// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todoReducer should return the initial state 1`] = `
Object {
  "todos": Array [],
}
`;
